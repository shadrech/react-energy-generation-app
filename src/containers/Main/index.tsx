import React from 'react';
import { EnergyContextProvider } from '../../contexts/EnergyContext';
import EnergyStatsWrapper from '../../components/EnergyStatsWrapper';

const Main: React.FunctionComponent = () => (
  <EnergyContextProvider>
    <EnergyStatsWrapper />
  </EnergyContextProvider>
)

export default Main
