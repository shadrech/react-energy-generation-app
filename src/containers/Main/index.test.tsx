import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Main from '.'

jest.mock('../../components/EnergyStatsWrapper', () => jest.fn().mockReturnValue(<div>EnergyStatsWrapper</div>))
jest.mock('../../contexts/EnergyContext', () => ({
  EnergyContextProvider: jest.fn().mockImplementation(props => <div data-test-id="context-provider" {...props} />)
}))

describe('<Main />', () => {

  afterEach(cleanup)

  it('renders as expected', () => {
    const { container } = render(<Main />)
    expect(container).toMatchSnapshot()
  })

})
