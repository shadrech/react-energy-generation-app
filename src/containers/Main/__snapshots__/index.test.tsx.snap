// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> renders as expected 1`] = `
<div>
  <div
    data-test-id="context-provider"
  >
    <div>
      EnergyStatsWrapper
    </div>
  </div>
</div>
`;
